Source/MVC6/Boilerplate.AspNetCore.TagHelpers/Twitter/Cards/TwitterCardApp.cs

```diff
@@ -105,5 +101,33 @@ public override void ToString(StringBuilder stringBuilder)
             stringBuilder.AppendMetaNameContentIfNotNull("twitter:app:url:googleplay", this.GooglePlayCustomUrlScheme);
             stringBuilder.AppendMetaNameContentIfNotNull("twitter:app:country", this.Country);
         }
+
+        /// <summary>
+        /// Checks that this instance is valid and throws exceptions if not valid.
+        /// </summary>
+        protected override void Validate()
+        {
+            base.Validate();
+
+            if (string.IsNullOrEmpty(this.TwitterSiteUsername))
+            {
+                throw new ArgumentNullException(nameof(this.TwitterSiteUsername));
+            }
+
+            if (string.IsNullOrEmpty(this.IPhone))
```

Comment:
This property is required in this case, I also reflected this in the [chart](https://github.com/ASP-NET-MVC-Boilerplate/Framework/issues/3).  You can confirm this requirement [here](https://dev.twitter.com/cards/types/app).